pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -0,0 +1,475 @@
+
+	//checking i finalizers exits in the dmsi cr adding if they dont
+	logger.Info("Checking for finalizers")
+	if utils.HasFinalizer(dmsi, deadMansSnitchFinalizer) == false {
+		log.Info(fmt.Sprint("Adding finalizer to DMSI Name: ", " DMSI Name: :"+dmsi.Name))
+		utils.AddFinalizer(dmsi, deadMansSnitchFinalizer)
+		err := r.client.Update(context.TODO(), dmsi)
+		if err != nil {
+			return err
+		}
+
+	}
+	logger.Info("DMSI finalizer created nothing to do here..: ")
+
+	return nil
+
+}
+
+// create snitch in deadmanssnitch.com with information retrived from dmsi cr as well as the matching cluster deployment
+func (r *ReconcileDeadmansSnitchIntegration) createSnitch(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, cd *hivev1.ClusterDeployment, dmsc dmsclient.Client) error {
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", cd.Name, "cluster-deployment.Namespace:", cd.Namespace)
+	snitchName := cd.Spec.ClusterName + "." + cd.Spec.BaseDomain + "-" + dmsi.Spec.SnitchNamePostFix
+	snitches, err := dmsc.FindSnitchesByName(snitchName)
+	logger.Info("Checking if snitch already exists SnitchName:", snitchName)
+	if err != nil {
+		return err
+	}
+
+	var snitch dmsclient.Snitch
+	if len(snitches) > 0 {
+		snitch = snitches[0]
+	} else {
+		newSnitch := dmsclient.NewSnitch(snitchName, dmsi.Spec.Tags, "15_minute", "basic")
+		snitch, err = dmsc.Create(newSnitch)
+		logger.Info("Creating snitch", snitchName)
+		if err != nil {
+			return err
+		}
+	}
+
+	ReSnitches, err := dmsc.FindSnitchesByName(snitchName)
+	if err != nil {
+		return err
+	}
+
+	if len(ReSnitches) > 0 {
+		if ReSnitches[0].Status == "pending" {
+			logger.Info("Checking in Snitch ...")
+			// CheckIn snitch
+			err = dmsc.CheckIn(snitch)
+			if err != nil {
+				logger.Error(err, "Unable to check in deadman's snitch", "CheckInURL", snitch.CheckInURL)
+				return err
+			}
+		}
+	} else {
+		logger.Error(err, "Unable to get Snitch by name")
+		return err
+	}
+	logger.Info("Snitch created nothing to do here.... ")
+	return nil
+}
+
+//Create secret containing the snitch url
+func (r *ReconcileDeadmansSnitchIntegration) createSecret(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, dmsc dmsclient.Client, cd hivev1.ClusterDeployment) error {
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", cd.Name, "cluster-deployment.Namespace:", cd.Namespace)
+	dmsSecret := cd.Spec.ClusterName + "-" + dmsi.Spec.SnitchNamePostFix + "-" + config.RefSecretPostfix
+	logger.Info("Checking if secret already exits")
+	err := r.client.Get(context.TODO(),
+		types.NamespacedName{Name: dmsSecret, Namespace: cd.Namespace},
+		&corev1.Secret{})
+	if errors.IsNotFound(err) {
+		logger.Info("Secret not found creating secret")
+		snitchName := cd.Spec.ClusterName + "." + cd.Spec.BaseDomain + "-" + dmsi.Spec.SnitchNamePostFix
```

Comment:
```suggestion
		snitchName := cd.Spec.ClusterName + "." + cd.Spec.BaseDomain
		if dmsi.Spec.SnitchNamePostFix != "" {
			snitchName += + "-" + dmsi.Spec.SnitchNamePostFix
		}
```
Since this is done in a few places it'd be best to have a utility function to encapsulate it.